components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,214 @@
+    { title: 'Today', period: { since: today, until: today } },
+    { title: 'Yesterday', period: { since: yesterday, until: yesterday } },
+    {
+      title: 'Last 7 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 7)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 30 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 30)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 90 days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 90)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Last 365 Days',
+      period: {
+        since: new Date(new Date().setDate(today.getDate() - 365)),
+        until: yesterday
+      }
+    },
+    {
+      title: 'Custom',
+      period: { since: yesterday, until: yesterday }
+    }
+  ];
+
+  const [popoverActive, setPopoverActive] = useState(false);
+  const [activeDateRange, setActiveDateRange] = useState(ranges[0]);
+  const [dateState, setDateState] = useState({
+    month: activeDateRange.period.since.getMonth(),
+    year: activeDateRange.period.since.getFullYear()
+  });
+
+  useEffect(() => {
+    setDateState({
+      month: activeDateRange.period.since.getMonth(),
+      year: activeDateRange.period.since.getFullYear()
+    });
+  }, [activeDateRange]);
+
+  const handleInputValueChange = useCallback((field, value) => {
+    if (isValidDate(value)) {
+      const newDate = new Date(value);
+      setActiveDateRange((prevState) => ({
+        ...prevState,
+        period: {
+          ...prevState.period,
+          [field]: newDate
+        }
+      }));
+    }
+  }, []);
+
+  function formatDate(date) {
+    return date.toISOString().split('T')[0];
+  }
+
+  function isValidDate(date) {
+    return !isNaN(new Date(date).getDate());
+  }
+
+  return (
+    <Box>
+      <Popover
+        active={popoverActive}
+        autofocusTarget='none'
+        preferredAlignment='left'
+        preferredPosition='below'
+        fluidContent
+        sectioned={false}
+        fullHeight
+        activator={
+          <Button icon={CalendarIcon} size='slim' onClick={() => setPopoverActive(!popoverActive)}>
+            {activeDateRange.title}
+          </Button>
+        }
+        onClose={() => setPopoverActive(false)}
+      >
+        <Box
+          style={{
+            paddingTop: '16px',
+            paddingLeft: '10px',
+            paddingRight: '10px',
+            paddingBottom: '16px'
+          }}
+        >
+          <Popover.Pane fixed>
+            <InlineGrid
+              columns={{ xs: '1fr', mdDown: '1fr', md: 'max-content max-content' }}
+              gap={0}
+            >
+              <Box
+                maxWidth='212px'
+                width='100%'
+                style={{
+                  paddingRight: '20px'
+                }}
+              >
+                <Scrollable style={{ height: '334px' }}>
```

Comment:
Couple small notes for mobile view, I'd just set this height to `auto` once it gets to a small enough screen size where everything stacks. It leaves a big padding at the bottom of the option list since the height is fixed.

Also I'd maybe just remove the arrow icon once the text fields stack, you can see in the screenshot it's kind of awkward to have there on mobile view.

![CleanShot 2024-05-08 at 23 57 52](https://github.com/RAAbbott/polaris-components/assets/41922804/20da6e98-f429-4c21-9e4e-38a8a0ac692b)